src/App.jsx

```diff
@@ -1,56 +1,128 @@
-import { initialColors } from "./lib/colors";
+import { initialThemes, initialColors } from "./lib/colors";
 import { uid } from "uid";
 import useLocalStorageState from "use-local-storage-state";
-import Color from "./Components/Color/Color";
-import ColorForm from "./Components/ColorForm/ColorForm";
+import Themes from "./Components/Themes/Themes";
 import "./App.css";
+import { ThemesMenu } from "./Components/ThemesMenu/ThemesMenu";
+import { useState } from "react";
 
 function App() {
+  const [themes, setThemes] = useLocalStorageState("themes", {
+    defaultValue: initialThemes,
+  });
+
+  const [currentTheme, setCurrentTheme] = useState(themes[0]);
+
   const [colors, setColors] = useLocalStorageState("colors", {
     defaultValue: initialColors,
   });
 
+  function updateCurrentTheme(updatedTheme) {
+    const updatedThemes = themes.map((theme) =>
+      theme.id === currentTheme.id ? updatedTheme : theme
+    );
+
+    setThemes(updatedThemes);
+    setCurrentTheme(updatedTheme);
+  }
+
   function handleAddColor(newColor) {
-    setColors([{ id: uid(), ...newColor }, ...colors]);
+    const updatedColor = { id: uid(), ...newColor };
+    setColors([updatedColor, ...colors]);
+    if (currentTheme.id === "t1") {
+      return;
+    } else {
+      const updatedTheme = {
+        ...currentTheme,
+        colors: [updatedColor, ...currentTheme.colors],
+      };
+      updateCurrentTheme(updatedTheme);
+    }
```

Comment:
and if you removed the colors state you can update the function: ` function handleAddColor(newColor) {

    const updatedColor = { id: uid(), ...newColor };

    const updatedTheme = {

      ...currentTheme,

      colors: [updatedColor, ...currentTheme.colors],

    };

    updateCurrentTheme(updatedTheme);`